crates/nu-command/tests/commands/unfold.rs

```diff
@@ -0,0 +1,64 @@
+use nu_test_support::{nu, pipeline};
+
+#[test]
+fn unfold_single_element_break() {
+    let actual = nu!("unfold 1 {|x| if $x == 3 { [$x] } else { [$x, ($x + 1)] }} | to nuon");
```

Comment:
Personally, I'm not in love with the syntax to using lists (as tuples) 

Not sure what I would like better though, records might be close:

`{ result: $x, state: ($x + 1) }`

Reply:
> Personally, I'm not in love with the syntax to using lists (as tuples)

@jakeswenson, I went back and forth here as well. To compare, here's a couple examples exploring how using records would look:
```nushell
unfold [0, 1] {|fib| {output: $fib.0, next: [$fib.1, ($fib.0 + $fib.1)]} } | first 10
```

```nushell
unfold 1 {|page|
  let resp = http get (
    {
      scheme: https,
      host: "api.github.com",
      path: "/repos/nushell/nushell/issues",
      params: {
	page: $page,
	per_page: $PAGE_SIZE
      }
    } | url join)

  if ($resp | length) < $PAGE_SIZE {
    {output: $resp}
  } else {
    {output: $resp, next: ($page + 1)}
  }
}
```

One reason I originally used a list are some corner cases around "invalid" returns:

If the generator returns a value that either isn't a record, or is a record that without an `output` key a runtime error would likely have to occur. 

```nushell
> unfold 1 {|x| if $x == 3 { $x } else { {output: $x, next: ($x + 1)} }}
<error stating invalid return. $x does not contain an `output` key>
```

If a `next` key existed, but no `output` key existed, I'm not sure whether that would indicate to stop, to output nothing and keep going, or to output `null` and keep going. Likewise, if any keys other than `output` or `next` are present in the return value, a runtime error would make sense.

--

All that being said, I agree that using lists in place of tuples isn't great. I'm open to changing to using records. It may be an improvement to the user experience and is also likely easier to document.

Reply:
After thinking it over, I've gone forward with using records for the return type, rather than lists.

I think the command is easier to understand after the change.